### ISSUE TITLE
Web UI dies when mesh.password is set

### ISSUE THREAD

**Issue by lil-cain** [prometheus/alertmanager#925]
When  I start two alert managers in a mesh with mesh.password set, and then visit <web.listen-adddress>/#/status, the alert manager who's webUI I have just visited immediately dies with the below stack trace

```
panic: assignment to entry in nil map

goroutine 15 [running]:
github.com/prometheus/alertmanager/vendor/github.com/weaveworks/mesh.makeLocalConnectionStatusSlice.func1(0xc420042f40)
        /go/src/github.com/prometheus/alertmanager/vendor/github.com/weaveworks/mesh/status.go:179 +0x42f
github.com/prometheus/alertmanager/vendor/github.com/weaveworks/mesh.(*connectionMaker).queryLoop(0xc420154150, 0xc42005d380)
        /go/src/github.com/prometheus/alertmanager/vendor/github.com/weaveworks/mesh/connection_maker.go:228 +0xf6
created by github.com/prometheus/alertmanager/vendor/github.com/weaveworks/mesh.newConnectionMaker
        /go/src/github.com/prometheus/alertmanager/vendor/github.com/weaveworks/mesh/connection_maker.go:75 +0x21a
```

The text "Network error" is displayed in the web UI.

I'm currently starting these with the following commandlines:

``` 
./alertmanager -mesh.listen-address 127.0.0.2:6783 -mesh.nickname peer1 -config.file simple.yml -log.level debug -web.listen-address 127.0.0.2:9003 -mesh.password omgwtflol

```

```
./alertmanager -mesh.listen-address 127.0.0.3:6783 -mesh.nickname peer2 -mesh.peer-id 68:94:23:5a:20:f1 -mesh.peer 127.0.0.2:6783 -config.file simple.yml -storage.path data1/ -web.listen-address 127.0.0.3:9003 -log.level debug -mesh.password omgwtflol
```

I'm using the below version, downloaded as a binary from prometheus.io

```
alertmanager, version 0.8.0 (branch: HEAD, revision: 74e7e48d24bddd2e2a80c7840af9b2de271cc74c)
  build user:       root@439065dc2905
  build date:       20170720-14:14:06
  go version:       go1.8.3
```

This does not occur if I run the same command without the -mesh.password argument.

**Comment by lil-cain** [comment]
This also occurs where I have >2 alert managers in a cluster.

**Comment by stuartnelson3** [comment]
Thanks for the report! We'll look at this soon.

**Comment by Kellel** [comment]
Looks like this is a bug in weaveworks/mesh

https://github.com/prometheus/alertmanager/blob/master/vendor/github.com/weaveworks/mesh/status.go#L170

Seems fixed upstream 
https://github.com/weaveworks/mesh/commit/71e3485f35d3d3339f543e4245192f7d73d339a3

**Comment by Kellel** [comment]
@lil-cain Would you mind testing to see if the patch resolves your issue?

**Comment by lil-cain** [comment]
Your branch is enough behind master for the test not to work immediately, but when i cherry-pick your fix onto master, it fixes the bug - thanks!

**Comment by Kellel** [comment]
Oh I'm sorry. I've rebased my pr.

**Comment by stuartnelson3** [comment]
Fixed by #960
